contracts/test/SetupPaymentsLib.t.sol

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Test.sol";
+import "../script/utils/SetupPaymentsLib.sol";
+import "../script/utils/CoreDeploymentLib.sol";
+import "../script/utils/HelloWorldDeploymentLib.sol";
+import "@eigenlayer/contracts/interfaces/IRewardsCoordinator.sol";
+import "@eigenlayer/contracts/interfaces/IStrategy.sol";
+import "../script/DeployEigenLayerCore.s.sol";
+import "../script/HelloWorldDeployer.s.sol";
+import {StrategyFactory} from "@eigenlayer/contracts/strategies/StrategyFactory.sol";
+
+contract TestConstants {
+    uint256 constant NUM_PAYMENTS = 8;
+    uint256 constant NUM_TOKEN_EARNINGS = 1;
+    uint256 constant TOKEN_EARNINGS = 100;
+
+    address RECIPIENT = address(1);
+    address EARNER = address(2);
+    uint256 INDEX_TO_PROVE = 0;
+}
+
+contract SetupPaymentsLibTest is Test, TestConstants {
```

Comment:
I want to move the hello world tests to be more realistic tests than what we do in some of our other repos.  Ideally, we should deploy the actual contracts for the hello world tests.

Since you added the RewardsCoordinator to be deployed in the deployment lib, it should be something like:

```
contract RewardsTest is Test {
    Quorum internal quorum;

    HelloWorldDeploymentLib.DeploymentData internal helloWorldDeployment;
    CoreDeploymentLib.DeploymentData internal coreDeployment;
    CoreDeploymentLib.DeploymentConfigData coreConfigData;

    ERC20Mock public mockToken;

    function setUp() public virtual {
        address proxyAdmin = UpgradeableProxyLib.deployProxyAdmin();
        coreConfigData =
           CoreDeploymentLib.readDeploymentConfigValues("test/mockData/config/core/", 1337); // TODO: Fix this to correct path
        coreDeployment = CoreDeploymentLib.deployContracts(proxyAdmin, coreConfigData);

        mockToken = new ERC20Mock();

        IStrategy strategy = addStrategy(address(mockToken)); // Similar function to HW_SM test using strategy factory
        quorum.strategies.push(StrategyParams({strategy: strategy, multiplier: 10_000}));

        helloWorldDeployment =
            HelloWorldDeploymentLib.deployContracts(proxyAdmin, coreDeployment, quorum);
        labelContracts(coreDeployment, helloWorldDeployment);
    }
}
```

Reply:
makes sense.  I was trying to test the integration library features in isolation but I will transition them to more integration style tests.